x-pack/plugins/security_solution/server/lib/timeline/routes/notes/get_notes.ts

```diff
@@ -139,11 +145,39 @@ export const getNotesRoute = (router: SecuritySolutionPluginRouter) => {
           const filterKueryNodeArray = [filterAsKueryNode];
 
           // retrieve all the notes created by a specific user
-          const userFilter = queryParams?.userFilter;
-          if (userFilter) {
-            filterKueryNodeArray.push(
-              nodeBuilder.is(`${noteSavedObjectType}.attributes.createdBy`, userFilter)
-            );
+          // the createdByFilter value is the uuid of the user
+          const createdByFilter = queryParams?.createdByFilter; // now uuid
+          if (createdByFilter) {
+            // because the notes createdBy property can be either full_name, email or username
+            // see pickSaveNote (https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/timeline/saved_object/notes/saved_object.ts#L302)
+            // which uses the getUserDisplayName (https://github.com/elastic/kibana/blob/main/packages/kbn-user-profile-components/src/user_profile.ts#L138)
+            const [_, { security }] = await startServices();
+            const users: UserProfile[] = await security.userProfiles.bulkGet({
+              uids: new Set([createdByFilter]),
+            });
+            // once we retrieve the user by the uuid we can search all the notes that have the createdBy property with full_name, email or username values
+            if (users && users.length > 0) {
+              const {
+                user: { email, full_name: fullName, username: userName },
+              } = users[0];
+              const createdByNodeArray = [];
+              if (fullName) {
+                createdByNodeArray.push(
+                  nodeBuilder.is(`${noteSavedObjectType}.attributes.createdBy`, fullName)
+                );
+              }
+              if (userName) {
+                createdByNodeArray.push(
+                  nodeBuilder.is(`${noteSavedObjectType}.attributes.createdBy`, userName)
+                );
+              }
+              if (email) {
+                createdByNodeArray.push(
+                  nodeBuilder.is(`${noteSavedObjectType}.attributes.createdBy`, email)
+                );
+              }
+              filterKueryNodeArray.push(nodeBuilder.or(createdByNodeArray));
+            }
```

Comment:
This is the logic I would like to have feedback on!
Thanks 😄

Reply:
I know I'm using the deprecated `security.userProfile` instead of the one from `core`, but the setup to be able to use the `core` is not in place and it felt like too many changes so late in the BC period...

Reply:
Spoke a bit offline, but I think this logic is more resilient 👍🏾.  Only main concern was making sure we always had access to the id of a user that would be searched, but that seems to be the case. Thanks!